src/main/scala/com/microsoft/hyperspace/actions/RefreshIncremental.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.actions
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.execution.datasources.{HadoopFsRelation, LogicalRelation, PartitioningAwareFileIndex}
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import com.microsoft.hyperspace.index._
+import com.microsoft.hyperspace.telemetry.{AppInfo, HyperspaceEvent, RefreshAppendActionEvent}
+
+class RefreshIncremental(
+    spark: SparkSession,
+    logManager: IndexLogManager,
+    dataManager: IndexDataManager)
+    extends RefreshActionBase(spark, logManager, dataManager) {
+  final override def op(): Unit = {
+    // TODO: The current implementation picks the number of buckets from session config.
+    //   This should be user-configurable to allow maintain the existing bucket numbers
+    //   in the index log entry.
+    write(spark, indexableDf, indexConfig)
+  }
+
+  private lazy val indexableDf = {
+    // Currently we only support creating an index on a LogicalRelation.
+    assert(previousIndexLogEntry.relations.size == 1)
```

Comment:
Thanks @sezruby , I think it will be wrong to rely on the assert in `relations`. Consider this scenario:

In future, say we start supporting multiple relations. But the current `RefreshIncremental` code depends on the fact that there is only one relation. If we remove the `assert` from here, we will forget to fix this class in the future.

If we leave `assert` in this class, it will fail in future when we start supporting multiple relations. This will make sure we don't forget  fixing this class.

Please let me know if this makes sense. Thanks.

Reply:
I am actually fine putting this assert in one place - `relations` instead of duplicating this in many places.

To me, we cannot blindly remove `assert` from `IndexLogEntry` without checking all the references. But if people think otherwise, I don't have a strong opinion on this.

Reply:
ok in that case I will move it. 
My only concern was it is prone to bugs and will depend on the original coders and memory to remember to fix this issue.